### ISSUE TITLE
Vterm does not have the same environment variable as native terminal on Mac OS Big Sur

### ISSUE THREAD

**Issue by XushanLu** [akermu/emacs-libvterm#528]
# This was originally reported to Spacemacs, but got diverted back here. 
# Description

The environment variable DYLD_LIBRARY_PATH is different from what I got from using terminal. That variable is unset in vterm but properly set in terminal. This caused my compiled executables not able to find the dynamic library with the following error:
`dyld: Library not loaded: @rpath/libmkl_intel_lp64.1.dylib`. 

The code runs perfectly fine within terminal.

# Reproduction Guide
- Open Emacs
- Open vterm 
- Type `echo $DYLD_LIBRARY_PATH`
- Open terminal
- Type `echo $DYLD_LIBRARY_PATH`

# Observed behavior
- Nothing prints within vterm
- Terminal prints the following
```shell
/opt/intel/oneapi/tbb/latest/env/../lib:/opt/intel/oneapi/mkl/latest/lib:/opt/intel/oneapi/ippcp/2021.2.0/lib:/opt/intel/oneapi/ipp/2021.2.0/lib:/opt/intel/oneapi/dnnl/2021.2.0/cpu_iomp/lib:/opt/intel/oneapi/dal/2021.2.0/lib:/opt/intel/oneapi/compiler/2021.2.0/mac/compiler/lib/intel64_mac:/opt/intel/oneapi/compiler/2021.2.0/mac/compiler/lib
```

# Expected behavior
- Both vterm and terminal return the same thing.

# System info
- OS: Mac OS Big Sur 11.03
- Emacs: 27.2
- Spacemacs: 0.300.0
- Spacemacs branch: develop
- Spacemacs layers:
```lisp
    dotspacemacs-configuration-layers
   '(vimscript javascript auto-completion better-defaults emacs-lisp helm html csv python ipython-notebook
     markdown org spell-checking syntax-checking version-control
     (shell :variables
            shell-default-height 30
            shell-default-shell 'vterm
            shell-default-term-shell "/usr/local/bin/zsh"
            shell-default-position 'bottom)
     (c-c++   :variables
              c-c++-default-mode-for-headers 'c++-mode
              c-c++-enable-clang-support t)
     pandoc ivy bibtex
     (latex :variables latex-build-command "LatexMk")
     eaf pdf
     (java :variables java-backend 'meghanada)
     gtags
     )
```
# Is this behavior caused by the Mac OS Security Integrity Protection? See [here](https://support.apple.com/en-ca/HT204899) and [here](https://stackoverflow.com/questions/60126159/how-to-set-ld-library-path-dyld-library-path-on-macos).

**Comment by wyuenho** [comment]
This has nothing to do vterm. Use [exec-path-from-shell](https://github.com/purcell/exec-path-from-shell)

**Comment by XushanLu** [comment]
Okay, I tried using exec-path-from-shell and I don't think it works. I also tried to set 'vterm-environment' and it does not work either. Is it possible that this is caused by Mac OS security integrity protection (SIP)?

**Comment by wyuenho** [comment]
You need to specify the environment variables you want to copy

https://github.com/purcell/exec-path-from-shell/blob/master/exec-path-from-shell.el#L85

**Comment by XushanLu** [comment]
Do I just directly modify that file directly? I did so and run `M-x exec-path-from-shell-initialize` and `M-x exec-path-from-shell-copy-env`. Then opened a new vterm buffer and tried `echo $DYLD_LIBRARY_PATH`. Nothing prints.

**Comment by XushanLu** [comment]
Now the file looks like 
```lisp
(defcustom exec-path-from-shell-variables
  '("PATH" "MANPATH" "DYLD_LIBRARY_PATH")
  "List of environment variables which are copied from the shell."
  :type '(repeat (string :tag "Environment variable"))
  :group 'exec-path-from-shell)
```

**Comment by wyuenho** [comment]
You either customize it with `M-x customize-option exec-path-from-shell-variables` or set it in your init file with `(setf exec-path-from-shell-variables '(...))`

**Comment by XushanLu** [comment]
I used the first method to do the customization. But I do not think I am getting anything by `echo $DYLD_LIBRARY_PATH`.

**Comment by wyuenho** [comment]
Did you restart emacs?

**Comment by XushanLu** [comment]
Yes. Restarted. Now the weird thing is that I cannot do the customization anymore if I `customize-option` because there is now no `exec-path-from-shell-variables` option anymore. But in my dotspacemacs file, there is this line
```lisp
 '(exec-path-from-shell-variables '("PATH" "MANPATH" "DYLD_LIBRARY_PATH"))
```
So, you don't think this is caused by Mac OS SIP?

**Comment by wyuenho** [comment]
>  I customize-option because there is now no exec-path-from-shell-variables option anymore

That usually means `exec-path-from-shell` wasn't loaded when you init. This package needs to be one of the first thing that loads after emacs has started.

**Comment by XushanLu** [comment]
Okay, I now added `(require 'exec-path-from-shell)` somewhere in my dotspacemacs file and it seems to be working, i.e., the package is now initialized and I am able to customize it again. Please see the following screenshot
![image](https://user-images.githubusercontent.com/7824380/120232523-49f4db80-c22e-11eb-9aef-8cb8ffa07c94.png)

But still nothing prints! Did I customize it correctly?

**Comment by wyuenho** [comment]
Actually, you might be right, `DYLD_LIBRARY_PATH` seems [rbenv and pyenv](https://github.com/rbenv/rbenv/issues/962) had similar issues. Best find a way so you don't have to use it. But in any case, the title of the ticket is not correct, the shell process spawned by vterm just inherits whatever emacs is using.

**Comment by XushanLu** [comment]
Okay, it seems that I really need to disable SIP if I want to use vterm the way I wanted. 

I don't think what I said in the title was wrong. My terminal gives me the correct variable values with `echo $DYLD_LIBRARY_PATH` but vterm does not. However, it is true that vterm picks up all environment variables generated by running `env` in my terminal.
